pinot-core/src/test/java/com/linkedin/pinot/query/aggregation/AggregationMVGroupByMVQueriesTest.java

```diff
@@ -0,0 +1,410 @@
+import com.linkedin.pinot.core.plan.Plan;
+import com.linkedin.pinot.core.plan.maker.InstancePlanMakerImplV2;
+import com.linkedin.pinot.core.plan.maker.PlanMaker;
+import com.linkedin.pinot.core.query.config.QueryExecutorConfig;
+import com.linkedin.pinot.core.query.reduce.BrokerReduceService;
+import com.linkedin.pinot.core.segment.creator.SegmentIndexCreationDriver;
+import com.linkedin.pinot.core.segment.creator.impl.SegmentCreationDriverFactory;
+import com.linkedin.pinot.pql.parsers.Pql2Compiler;
+import com.linkedin.pinot.segments.v1.creator.SegmentTestUtils;
+import com.linkedin.pinot.util.TestUtils;
+
+
+public class AggregationMVGroupByMVQueriesTest {
+
+  private final static Logger LOGGER = LoggerFactory.getLogger(AggregationMVGroupByMVQueriesTest.class);
+
+  private final static String AVRO_DATA = "data/test_data-mv.avro";
+  private final static File INDEXES_DIR = new File(FileUtils.getTempDirectory() + File.separator
+      + "AggregationMVGroupByMVQueriesTest");
+  private final static Pql2Compiler PQL2_COMPILER = new Pql2Compiler();
+  private final static List<SegmentDataManager> INDEX_SEGMENT_LIST = new ArrayList<SegmentDataManager>();
+  private final int NUM_SEGMENTS = 20;
+  private static PlanMaker INSTANCE_PLAN_MAKER;
+
+  @BeforeClass
+  public void setup() throws Exception {
+    INSTANCE_PLAN_MAKER = new InstancePlanMakerImplV2(new QueryExecutorConfig(new PropertiesConfiguration()));
+    setupSegmentList(NUM_SEGMENTS);
+  }
+
+  @AfterClass
+  public void tearDown() {
+    if (INDEXES_DIR.exists()) {
+      FileUtils.deleteQuietly(INDEXES_DIR);
+    }
+    for (SegmentDataManager segmentDataManager : INDEX_SEGMENT_LIST) {
+      segmentDataManager.getSegment().destroy();
+    }
+    INDEX_SEGMENT_LIST.clear();
+  }
+
+  private void setupSegmentList(int numberOfSegments) throws Exception {
+    final String filePath = TestUtils.getFileFromResourceUrl(getClass().getClassLoader().getResource(AVRO_DATA));
+
+    if (INDEXES_DIR.exists()) {
+      FileUtils.deleteQuietly(INDEXES_DIR);
+    }
+    INDEXES_DIR.mkdir();
+
+    for (int i = 0; i < numberOfSegments; ++i) {
+      final File segmentDir = new File(INDEXES_DIR, "segment_" + i);
+
+      final SegmentGeneratorConfig config =
+          SegmentTestUtils.getSegmentGenSpecWithSchemAndProjectedColumns(
+              new File(filePath),
+              segmentDir,
+              "daysSinceEpoch",
+              TimeUnit.DAYS,
+              "test");
+
+      final SegmentIndexCreationDriver driver = SegmentCreationDriverFactory.get(null);
+      driver.init(config);
+      driver.build();
+
+      LOGGER.debug("built at : {}", segmentDir.getAbsolutePath());
+      INDEX_SEGMENT_LIST.add(
+          new OfflineSegmentDataManager(
+              ColumnarSegmentLoader.load(
+                  new File(segmentDir, driver.getSegmentName()),
+                  ReadMode.heap)));
+    }
+  }
+
+  @Test
+  public void testSumAggrMVGroupBySV() throws Exception {
```

Comment:
Just my preference: I'd rather generate a segment with specific data, so I can also generate expected results programatically, instead of hard-coding.